textfields [1 of 3]: add text into speech bubble; also add rich text example

mimecuvalo opened the pull request:
This is the first of three textfield changes. This starts with making the speech bubble actually have text. Also, it creates a TipTap example and how that would be wired up.

🎵 this is dangerous, I walk through textfields so watch your head rock 🎵 

### Change Type

- [x] `minor` — New feature

### Release Notes

- Refactor textfields be composable/swappable.

SomeHats commented:
If we're going to add these examples, I'd like them to demonstrate a slightly more complete idea of what adding text to a shape might involve and the type of bugs you might run into e.g.

How should the shape change as the text grows?
![Kapture 2024-03-04 at 12 24 42](https://github.com/tldraw/tldraw/assets/1489520/057c1101-6c74-4ea4-8b0f-dfd7f6de1e35)

Focus management is hard
![Kapture 2024-03-04 at 12 29 57](https://github.com/tldraw/tldraw/assets/1489520/edc804a2-bd80-4442-b692-455424405b5d)

steveruizok commented on packages/tldraw/src/lib/shapes/text/TextArea.tsx:
```diff
@@ -0,0 +1,49 @@
+	handleInputPointerDown,
+	handleDoubleClick,
+}: {
+	rInput: React.RefObject<HTMLTextAreaElement>
```
forward ref instead here?

steveruizok replied:
(I mean they're taking it out in a few months anyway, but for now!)

mimecuvalo replied:
hah doh, not sure what i was smoking here - reworked!

steveruizok commented:
Not sure what the priorities are set to, but could we push the more complex speech bubble example down?

<img width="419" alt="image" src="https://github.com/tldraw/tldraw/assets/23072548/c42489d8-4986-4204-83b8-804dbf752c84">

mimecuvalo commented:
- fixed up growing for shapes. the rich text one isn't 100% perfect for every edge case but it's meant to be an example so I figure it doesn't have to reach 100% there, it's meant to be a demo.
- fixed up focus management, good flag @SomeHats 
- fixed up the forwardRef, doh - thx @steveruizok 
- changed priority of examples

mimecuvalo commented:
The refactor work in this PR would help support https://github.com/tldraw/tldraw/pull/3132 (another reason I was working on textfields)
We should land this first PR at least, please! @steveruizok it's waiting your approval ✅ - thx!

steveruizok commented:
Let's split this up!

Let's land:
- the changes that turn our <textareas> into a TextArea component
- the simple example

Let's not land:
- the changes to TextEditor
- the rich text example
